test/cli/daemon.js

```diff
@@ -265,4 +282,34 @@ describe('daemon', () => {
       expect(err.stdout).to.include(`Node.js version: ${process.versions.node}`)
     }
   })
+
+  it('should init by default if not already', async function () {
```

Comment:
How does this test assert that the daemon auto-initialised it's own repo?

Reply:
ipfs daemon cmd throws if there is no repo initialised. So if this test passes we should be good.

Reply:
It should also assert that the repo was created after the command has run, otherwise it doesn't actually test anything other than that the daemon process was killed.  Check the repo directory exists or something like that.

Reply:
My reasoning is to not over test stuff, here we are testing the `daemon` command and logic, `init` was added to that logic but we should only test that it gets ran not that `init` works correctly we have other tests for that.
This test will fail if init isn't ran or if init fails internally, which is the scope i wanted when i wrote the test. 
Makes sense ?

Reply:
I guess its a bit hidden because init is on by default and if init doesn't work daemon also fails completely but still we have other tests that make sure init creates a repo.

Reply:
If this test doesn't test anything not covered by other tests it should be removed.

If it does test something not covered by other tests, it should contain assertions to prove the functionality being tested works as expected.

Too many of our tests don't actually test anything, they just ensure that nothing on the happy path explodes so aren't all that useful.

Reply:
@achingbrain is right. Tests should be both valuable and informative. If it requires one to know that the test is testing for the program not to blow up in some weird situation, at least that needs to be well documented and not just live in one persons brain.

Reply:
the assertion is this `if (stdout.includes('Daemon is ready')) {` i agree this isn't the best way to do things and i will improve this in a near future, but this is the way we assert a daemon ran successfully in the others tests. 

this is the first test to work without calling `ipfs(init)`.

either i remove the test or change the name to something more informative like 'should start the daemon and init automatically without calling init first' or something else but i don't think testing for the repo is useful here. 
im open to suggestions

Reply:
Suggestions were:
- Document it with code comments
- Assert that the repo was created as part of the test